enhancements/api-review/add-new-CRD-ImageSourceDigestPolicy-and-ImageSourceTagPolicy-to-config.openshift.io.md

```diff
@@ -0,0 +1,400 @@
+	// may impact the exact order mirrors are contacted in, or some mirrors may be contacted
+	// in parallel, so this should be considered a preference rather than a guarantee of ordering.
+	// "mirrors" ueses one of the following formats:
+	// host[:port]
+	// host[:port]/namespace[/namespace…]
+	// host[:port]/namespace[/namespace…]/repo
+	// for more information about the format, see the document about the location field:
+	// https://github.com/containers/image/blob/main/docs/containers-registries.conf.5.md#choosing-a-registry-toml-table
+	// +optional
+	// +listType=set
+	Mirrors []Mirrors `json:"mirrors,omitempty"`
+}
+
+// +kubebuilder:validation:Pattern=`^((?:[a-zA-Z0-9]|[a-zA-Z0-9][a-zA-Z0-9-]*[a-zA-Z0-9])(?:(?:\.(?:[a-zA-Z0-9]|[a-zA-Z0-9][a-zA-Z0-9-]*[a-zA-Z0-9]))+)?(?::[0-9]+)?)(?:(?:/[a-z0-9]+(?:(?:(?:[._]|__|[-]*)[a-z0-9]+)+)?)+)?$`
+type Mirror string
+```
+
+An example ImageSourceTagPolicy file will look like:
+```yaml
+apiVersion: config.openshift.io/v1
+kind: ImageSourceTagPolicy
+metadata:
+  name: ubi8repo
+spec:
+  repositoryTagMirrors: 
+  - mirrors:
+    - example.io/example/ubi-minimal
+    source: registry.access.redhat.com/ubi8/ubi-minimal
+```
+
+The ImageSourceTagPolicy object will lead to the creation of a drop-in file at `/etc/containers/registries.conf` by MCO containerruntime controller that no
+digest only limitations on this mirrors.
+
+### User Stories
+
+#### As a user, I would like to pull images from mirror using tag, without digest references
+
+The user need to define multiple mirror configurations, used by apps/manifests that don't use digests to pull
+the images when working with disconnected environments or pulling from registries that act as transparent pull-through proxy cache.
+
+For user on a cluster that alreay supports ImageSourceTagPolicy and ImageSourceDigestPolicy, the user can pull image without digest by configuring mirrors using `ImageSourceTagPolicy` CRD.
+For user upgraded to a cluster suppports `ImageSourceTagPolicy` CRD, if they had ICSP objects, the ICSP objects still take
+effect. The user need to configure mirrors using `ImageSourceTagPolicy` CRD on the upgraded cluster to use tags.
+The MCO will create the ImageSourceTagPolicy object. Once this is done, the images can be pulled from the mirrors without the digest referenced.
+
+#### As a user using ICSP, I would like to use ICSP pull images using digest by default from mirrors
+
+The user can still define ICSP CR before its deprecation. After the deprecation of ICSP, the uset need to use
+`ImageSourceDigestPolicy` for digest pull specification requirement.
+
+### API Extensions
+- Adds new CRD ImageSourceDigestPolicy to config.openshift.io/v1.
+- Adds new CRD ImageSourceTagPolicy to config.openshift.io/v1
+- Impacts on existing resources(The impacts are in the "Operational Aspects
+of API Extensions" section)
+  - Components currently using ImageContentSourcePolicy should upgrade to use ImageSourceDigestPolicy.
```

Comment:
can you double chck the spacing on this "-" it is nested under the prev bullet, is that correct?

Reply:
Yes, it's correct